packages/headless/src/controllers/core/context/headless-core-context.ts

```diff
@@ -119,21 +124,41 @@ export function buildCoreContext(
         values: getState().context.contextValues,
       };
     },
-
     set(context: ContextPayload) {
       dispatch(setContext(context));
     },
-
-    add(contextKey: string, contextValue: ContextValue) {
-      dispatch(addContext({contextKey, contextValue}));
-    },
-
-    remove(key: string) {
-      dispatch(removeContext(key));
-    },
+    ...(getState().configuration.analytics.analyticsMode === 'legacy'
+      ? legacyCoreContext(dispatch)
+      : nextCoreContext(dispatch)),
   };
 }
 
+const legacyCoreContext = (dispatch: Dispatch<AnyAction>) => ({
+  add(contextKey: string, contextValue: ContextValue) {
+    dispatch(addContext({contextKey, contextValue}));
+  },
+
+  remove(key: string) {
+    dispatch(removeContext(key));
+  },
+});
+
+const nextCoreContext = (dispatch: Dispatch<AnyAction>) => ({
+  add(contextKey: string, contextValue: ContextValue) {
+    if (isReservedContextKey(contextKey)) {
+      throw new ReservedContextKeyError(contextKey);
+    }
+    dispatch(addContext({contextKey, contextValue}));
+  },
+
+  remove(key: string) {
+    if (isReservedContextKey(key)) {
+      throw new ReservedContextKeyError(key);
+    }
+    dispatch(removeContext(key));
+  },
+});
```

Comment:
in add the key is called `contextKey`.
in remove the key is called `key`.

Reply:
Sure
```suggestion
const nextCoreContext = (dispatch: Dispatch<AnyAction>) => ({
  add(contextKey: string, contextValue: ContextValue) {
    if (isReservedContextKey(contextKey)) {
      throw new ReservedContextKeyError(contextKey);
    }
    dispatch(addContext({contextKey, contextValue}));
  },

  remove(contextKey: string) {
    if (isReservedContextKey(contextKey)) {
      throw new ReservedContextKeyError(contextKey);
    }
    dispatch(removeContext(contextKey));
  },
});
```